docs/csharp/language-reference/keywords/where-generic-type-constraint.md

```diff
@@ -10,57 +10,68 @@ f1_keywords:
-  
- [!code-csharp[csrefKeywordsContextual#7](../../../csharp/language-reference/keywords/codesnippet/CSharp/where-generic-type-constraint_2.cs)]  
-  
- The `new()` constraint appears last in the `where` clause.  
-  
- With multiple type parameters, use one `where` clause for each type parameter, for example:  
-  
- [!code-csharp[csrefKeywordsContextual#8](../../../csharp/language-reference/keywords/codesnippet/CSharp/where-generic-type-constraint_3.cs)]  
-  
- You can also attach constraints to type parameters of generic methods, like this:  
-  
-```csharp  
-public bool MyMethod<T>(T t) where T : IMyInterface { }  
-```  
-  
- Notice that the syntax to describe type parameter constraints on delegates is the same as that of methods:  
-  
-```csharp  
-delegate T MyDelegate<T>() where T : new()  
-```  
-  
- For information on generic delegates, see [Generic Delegates](../../../csharp/programming-guide/generics/generic-delegates.md).  
-  
- For details on the syntax and use of constraints, see [Constraints on Type Parameters](../../../csharp/programming-guide/generics/constraints-on-type-parameters.md).  
-  
-## C# Language Specification  
- [!INCLUDE[CSharplangspec](~/includes/csharplangspec-md.md)]  
-  
-## See Also  
+> For more information on the where clause in a query expression, see [where clause](where-clause.md).
+
+In addition to interface constraints, a `where` clause can include a base class constraint, which states that a type must have the specified class as a base class (or be that class itself) in order to be used as a type argument for that generic type. If such a constraint is used, it must appear before any other constraints on that type parameter. Some types are disallowed as a base class constraint. These restrictions were relaxed beginning with C# 7.3. The following examples shows the types that can now be specified as a base class:
+
+[!code-csharp[using an interface constraint](../../../../samples/snippets/csharp/keywords/GenericWhereConstraints.cs#2)]
+
+The `where` clause can specify that the type is a `class` or a `struct`. The `struct` constraint removes the need to specify a base class constraint of `System.ValueType`. The `System.ValueType` type may not be used as
+a base class constraint. The following example shouws both the `class` and `struct` constraints:
+
+[!code-csharp[using the class and struct constraints](../../../../samples/snippets/csharp/keywords/GenericWhereConstraints.cs#3)]
+
+The `where` clause may also include an `unmanaged` constraint. The unmanaged
```

Comment:
Are the line breaks for this paragraph intentional?
Sentences above (ex. line 25) fit on one line, but here some sentences contain line breaks.